src/opera/test/pge/dswx_hls/test_dswx_hls_pge.py

```diff
@@ -369,6 +388,34 @@ def test_dswx_product_metadata_collection(self):
         self.assertEqual(output_product_metadata['yCoordinates']['size'], 3660)
         self.assertEqual(output_product_metadata['yCoordinates']['spacing'], 30)
 
+    @patch.object(opera.util.img_utils, "gdal", CustomMockGdal_InvalidLandsat)
+    def test_dswx_validate_expected_input_platforms_landsat(self):
+        """Test exceptions raised for Landsat 7 input data"""
+
+        # Create a filename that will trigger a metadata check
+        input_file = os.path.join(self.input_dir, "HLS.L30.T22VEQ.2021248T143156.v2.0.VAA.tif")
+        os.system(f"touch {input_file}")
+
+        runconfig_path = join(self.data_dir, 'test_dswx_hls_config.yaml')
+
+        pge = DSWxHLSExecutor(pge_name="DSWxPgeTest", runconfig_path=runconfig_path)
+        with self.assertRaises(Exception) as context:
```

Comment:
Ensuring we get an Exception for these test cases is definitely what we want, but it also carries the risk of causing the test to pass if code throws any kind of unrelated Exception, essentially masking the original issue. The workaround I've tried to do for this is to add another check on the contents of the `logger` to ensure we got the expected error message (or at least an easy-to-search-for fragment). That way, we can be sure the test actually triggered the error we expected.

Lines [167-181 ](https://github.com/nasa/opera-sds-pge/blob/develop-R2/src/opera/test/pge/dswx_hls/test_dswx_hls_pge.py#L167) here are a good example you can copy pretty much wholesale, and just add the log fragment to search for.